File: computer-based-test/app/Models/guru.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class guru extends Model
{

    protected $table = 'guru'; 

    protected $primaryKey = 'id_guru';

    protected $fillable = [
        'id_guru',
        'nama_guru',
        'nip',
        'status',
        'id_user',
        'id_operator',
    ];

    public function user(){
        return $this->belongsTo(User::class,'id_user');
    }

    public function operator() {
        return $this->belongsTo(operator::class);
    }

    public function kursus(){
        return $this->hasMany(kursus::class);
    }

    public function guru(){
        return $this->hasMany(latihan::class);
    }

    public function ujian(){
        return $this->hasMany(ujian::class);
    }
}

File: cbt-api/entity/guru.go
package entity

import "time"

type Guru struct {
    IdGuru     uint64    `gorm:"primary_key;autoIncrement" json:"id_guru"`
    NamaGuru   string    `gorm:"type:varchar(255);not null" json:"nama_guru"`
    NIP        int       `gorm:"type:int(11);not null" json:"nip"`
    Status     string    `gorm:"type:enum('Aktif', 'Tidak Aktif');not null" json:"status"`
    IdUser     uint64   `gorm:"type:bigint(20) unsigned" json:"id_user"` // Foreign Key ke Users
    IdOperator uint64    `gorm:"type:bigint(20) unsigned" json:"id_operator"` // Foreign Key ke Operators
    CreatedAt  time.Time `gorm:"type:timestamp;default:CURRENT_TIMESTAMP" json:"created_at"`
    UpdatedAt  time.Time `gorm:"type:timestamp;default:CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP" json:"updated_at"`

    // Foreign key relations
    Users      Users     `gorm:"foreignkey:id_user;constraint:onUpdate:CASCADE, onDelete:CASCADE" json:"user"`
    Operator   Operator  `gorm:"foreignkey:id_operator;constraint:onUpdate:CASCADE, onDelete:CASCADE" json:"operator"`
}

func (Guru) TableName() string {
    return "guru" // Nama tabel di database
}
